infra/main.bicep

```diff
@@ -218,6 +267,10 @@ var tags = { 'azd-env-name': environmentName }
 var rgName = 'rg-${environmentName}'
 var keyVaultName = 'kv-${resourceToken}'
 
+var isGPT4 = azureOpenAIModel == 'gpt-4'
+var isVision = azureOpenAIModelVersion == 'vision-preview'
```

Comment:
Can isVision ever be true when isGPT4 is false?

Also, I'm thinking, to keep things simple, can we not force azureOpenAIModelVersion to be vision-preview when using GPT4? Is there a reason a user would definitely not want to use vision-preview when using GPT4? 

This way, we can just check if GPT4 is being used and assume vision is included.

Reply:
> Can isVision ever be true when isGPT4 is false?

Not currently. But this may change in the future.

> Also, I'm thinking, to keep things simple, can we not force azureOpenAIModelVersion to be vision-preview when using GPT4? Is there a reason a user would definitely not want to use vision-preview when using GPT4?

I don't think we want to force vision if gpt-4 is being used, users may want to use different models or may be restricted based on region. https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#model-summary-table-and-region-availability